src/client/game.ts

```diff
@@ -104,79 +108,74 @@ export default class ClientGame extends GameLoop {
   }
 
   serverMsg(data: any): void {
-    if (!this.running || this.my_id === undefined) return;
+    if (!this.running) return;
 
     const message = deserializeSTC(data);
 
     if (this.my_id in message.inputAck) {
       this.inputHistory.discard_front_until(message.inputAck[this.my_id]);
     }
 
-    this.update_players(message.state);
-    this.update_enemies(message.state);
+    const prevState = this.states.last_elem();
+    const newState = message.state;
 
-    this.states.reset(message.state, message.stateNum);
+    this.update_player_sprites(prevState, newState);
+    this.update_enemy_sprites(prevState, newState);
+
+    this.states.reset(newState, message.stateNum);
   }
 
-  update_players(state: State): void {
+  update_player_sprites(prevState: State | undefined, newState: State): void {
     // spawn new players
-    for (const player of Object.values(state.players)) {
+    for (const player of Object.values(newState.players)) {
       if (this.player_list[player.id] === undefined) {
-        // the positions of the players do not matter since they will be corrected by the server
-        // TODO: we should probably think of a better solution in the future
-        this.add_character(
-          PLAYER_SPAWN_X_MIN,
-          PLAYER_SPAWN_Y_MIN,
-          PLAYER_SCALE,
-          PLAYER_SPRITE,
-          player.id,
-        );
-
-        if (player.id === this.my_id && this.my_id !== undefined) {
+        this.add_character(0, 0, PLAYER_SIZE, PLAYER_SPRITE, player.id);
```

Comment:
borde sätta positionen korrekt, alternativ göra om add_character så att den inte tar in en position